src/ansi-c/c_nondet_symbol_factory.cpp

```diff
@@ -53,24 +61,95 @@ class symbol_factoryt
     const typet &allocate_type,
     const bool static_lifetime);
 
+  void gen_array_allocation(
+    code_blockt &assignments,
+    const exprt &array_expr,
+    const exprt &size);
+
+  /// Generate a function that behaves like malloc from our stdlib
+  /// implementation
+  /// \param malloc_symbol_name The name of the malloc function
+  const symbolt &gen_malloc_function(const irep_idt &malloc_symbol_name);
+
   void gen_nondet_init(
     code_blockt &assignments,
     const exprt &expr,
     const std::size_t depth = 0,
     recursion_sett recursion_set = recursion_sett());
 
+  void gen_nondet_array_member_initialization(
+    code_blockt &assignments,
+    const exprt &array,
+    const exprt &array_size,
+    std::size_t depth,
+    const recursion_sett &recursion_set);
+
+  void gen_nondet_string_member_initialization(
+    code_blockt &assignments,
+    const exprt &array,
+    const exprt &array_size,
+    std::size_t depth,
+    const recursion_sett &recursion_set);
+
 private:
-  /// Generate initialisation code for each array element
-  /// \param assignments: The code block to add code to
-  /// \param expr: An expression of array type
-  /// \param depth: The struct recursion depth
-  /// \param recursion_set: The struct recursion set
-  /// \see gen_nondet_init For the meaning of the last 2 parameters
+  /// Add a new variable symbol to the symbol table
+  /// \param type: The type of the new variable
+  /// \param prefix: This forms the first part of the parameter,
+  ///   for debugging purposes must be a valid identifier prefix
+  /// \return A reference to the newly created symbol table entry
+  symbolt &new_tmp_symbol(const typet &type, const std::string &prefix);
```

Comment:
This can now be replaced by `allocate_objectst::allocate_automatic_local_object()`